src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc

```diff
@@ -402,7 +422,9 @@ BtreeLBAManager::_alloc_extents(
 	    rets.emplace_back(iter.get_pin(c));
```

Comment:
See another comment. I think in order to make sure the comment "// once assigned, the allocation to key must be exact and successful" is accurate, we should move the assertions
```c++
assert(mapping->get_key() == alloc_info.key);
assert(mapping->get_raw_val().get_laddr() ==
       alloc_info.val.get_laddr());
assert(mapping->get_length() == alloc_info.len);
```
into `_alloc_extents()`.